logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -185,6 +193,19 @@ abstract class UserSessionScopeCommon internal constructor(
 
     private val clientIdProvider = CurrentClientIdProvider { clientId() }
 
+    // TODO: make atomic
+    // val _teamId: Atomic<Either<CoreFailure, TeamId?>> = Atomic(Either.Left(CoreFailure.Unknown(Throwable("NotInitialized"))))
+    private var _teamId: Either<CoreFailure, TeamId?> = Either.Left(CoreFailure.Unknown(Throwable("NotInitialized")))
+
+    private suspend fun teamId(): Either<CoreFailure, TeamId?> =
+        if (_teamId.isRight()) _teamId else {
+            userRepository.userById(userId).map {
+                _teamId = Either.Right(it.teamId)
+                it.teamId
+            }
+        }
+    private val selfTeamId = SelfTeamIdProvider { teamId() }
```

Comment:
nice one I think we could create our own Delegated property access for those thing

Reply:
I think we could extract these to a separate class instead of having all of these definitions inside `UserSessionScope`.

This way we can actually test them. Which is also missing.